internal/core/workers/operation_execution_worker/operation_execution_worker.go

```diff
@@ -89,14 +89,25 @@ func (w *OperationExecutionWorker) Start(ctx context.Context) error {
 
 		if op.Status != operation.StatusPending {
 			loopLogger.Warn("operation is at an invalid status to proceed")
+
+			err = w.operationManager.AppendOperationEventToExecutionHistory(ctx, op, "Operation is at an invalid status to proceed")
+			if err != nil {
+				return fmt.Errorf("Error appending operation event to execution history: %w", err)
+			}
 			continue
 		}
 
 		executor, hasExecutor := w.executorsByName[def.Name()]
 		if !hasExecutor {
 			loopLogger.Warn("operation definition has no executor")
+
 			w.evictOperation(ctx, loopLogger, op)
 			reportOperationEvicted(w.schedulerName, op.DefinitionName, LabelNoOperationExecutorFound)
+
+			err = w.operationManager.AppendOperationEventToExecutionHistory(ctx, op, "Operation definition has no executor")
+			if err != nil {
+				return fmt.Errorf("Error appending operation event to execution history: %w", err)
```

Comment:
In this case I think we should not return, because returning here means closing the worker. In these scenarios, I think we should log ERROR and continue.